### PR TITLE
[BE] assign a random library color to all user_created libraries

### PR DESCRIPTION
**Pull request by ahsu1230**
(1) when a library is created, assign a random library color (based on the current color selection on the front-end)
(2) there's also an endpoint to be called that will paginate between all `user_created` libraries and each saving a randomly chosen color. This endpoint will be removed later once all non-system libraries have a color

@andrewconner @2is10

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/model/Library.scala [hunk 1]
```diff
@@ -254,4 +256,18 @@ object HexColor {
...
   def apply(hex: String): HexColor = new HexColor(hex.toLowerCase)
+
+  val allColors = Seq(
+    HexColor("#C764A2"), // magenta (pink)
+    HexColor("#E35957"), // red
+    HexColor("#FF9430"), // orange (a bit darker)
+    HexColor("#2EC89A"), // indigo (light green)
+    HexColor("#3975BF"), // blue
+    HexColor("#955CB4"), // purple
+    HexColor("#FAB200")) // light orange (more yellow-ish)
```
Comment (2is10) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:267]: How confident are we of these colors? I saw a mobile color picker design from Teehan (looking for it now) and I’m not sure it had exactly these colors.

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes [hunk 2]
```diff
@@ -354,6 +354,8 @@ POST    /site/libraries/add         @com.keepit.controllers.website.LibraryContr
 POST    /site/libraries/copy        @com.keepit.controllers.website.LibraryController.copyKeeps()
 POST    /site/libraries/move        @com.keepit.controllers.website.LibraryController.moveKeeps()
 
+POST    /site/libraries/colorAll    @com.keepit.controllers.website.LibraryController.colorAllUserLibraries()
```
Comment (2is10) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes]: This should be an admin endpoint in an admin controller, no?

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/LibraryController.scala [hunk 3]
```diff
@@ -48,6 +48,28 @@ class LibraryController @Inject() (
...
+        }
+      }
+      c += 1
+      paginator = Paginator(c, pageSize)
+      libs = db.readOnlyMaster { implicit s =>
+        libraryRepo.pageByKind(LibraryKind.USER_CREATED, paginator)
+      }
+    }
+    NoContent
+  }
```
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/LibraryController.scala]: If we end up needing to run this more than once for whatever reason, it’d be more efficient to filter out libraries that already have colors at the database level.

**Comment by 2is10** [comment]
:+1:

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/website/LibraryControllerTest.scala [hunk 4]
```diff
@@ -65,6 +67,44 @@ class LibraryControllerTest extends Specification with ShoeboxTestInjector {
...
+          library().withUser(user5.id.get).withKind(LibraryKind.SYSTEM_SECRET).secret().saved
+
+          libraryRepo.count === 9
+          val allLibs = libraryRepo.all
+          allLibs.count(_.color.isEmpty) === 8
+          allLibs.count(_.color.nonEmpty) === 1
+          user1
+        }
+
+        val libraryController = inject[AdminLibraryController]
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/LibraryControllerTest.scala:93]: It's in `AdminLibraryController`, but there's no `AdminLibraryControllerTest` so I'll put the test in here for now and remove it later!

Comment (2is10) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/LibraryControllerTest.scala:93]: OK by me

File: kifi-backend/modules/common/app/com/keepit/model/Library.scala (hunk 1 above)
Comment (ahsu1230) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:267]: Just confirmed with Mark that these library colors are finalized!

File: kifi-backend/modules/common/app/com/keepit/model/Library.scala [hunk 5]
```diff
@@ -254,4 +256,18 @@ object HexColor {
 
   val regex = "^#[0-9a-f]{6}$".r
   def apply(hex: String): HexColor = new HexColor(hex.toLowerCase)
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:258]: You may want to match just on the allowed colors, so a client can't pick a non-"approved" color.

**Pull request merged**